config/envprovider.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package config // import "go.opentelemetry.io/contrib/config"
+
+import (
+	"os"
+	"regexp"
+	"strings"
+)
+
+const ValidationPattern = `^[a-zA-Z_][a-zA-Z0-9_]*$`
+
+var validationRegexp = regexp.MustCompile(ValidationPattern)
+
+func replaceEnvVar(uri string) []byte {
+	envVarName, defaultValuePtr := parseEnvVarURI(uri)
+	if !validationRegexp.MatchString(envVarName) {
+		return nil
+	}
+
+	val, exists := os.LookupEnv(envVarName)
```

Comment:
From https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#parsing-empty-value

> The SDK MUST interpret an empty value of an environment variable the same way as when the variable is unset.

Reasons are here: https://github.com/open-telemetry/opentelemetry-specification/pull/2045. Maybe I should add the reasons to spec as well, but it was I think my first contribution to the specification.